valora-inc/wallet
# fix(pfp): feature flag for "step two" of PFP

### Description

Currently "step 2" of the link bank account flow is not sealed off by default for users who have completed KYC with Persona. This is a problem because the Plaid flow is still under development.

This adds a feature flag to make sure the button is disabled until we are done with step 2 of the PFP (Persona, Finclusive, Plaid) integration.

### Other changes

na

### Tested

added automated test coverage

manual test looks good. With firebase remote config `linkBankAccountStepTwoEnabled` turned off:
![simulator_screenshot_DABFECDB-42F6-47CC-9CC7-5B4DDDD63165](https://user-images.githubusercontent.com/7979215/151636322-dec805f5-410f-4cd8-8f8c-e099802f4835.png)

and with `linkBankAccountStepTwoEnabled` turned on:
![simulator_screenshot_EA70F8F9-16E4-4109-BD1D-1883ADC93177](https://user-images.githubusercontent.com/7979215/151636444-6de18caf-6950-4228-8eb9-f30cc0bdb6ad.png)


### How others should test

Since connecting your bank account is not complete, you should expect the "step two" section of the "link your bank account" settings page to have a button that is disabled, even after you verify your identity with Persona.

To test:
- with an account on alfajores with a verified phone number:
- go to settings
- tap "Link Bank Accounts"
- complete Persona flow ("step 1")
- verify that "coming soon!" button is disabled (greyed out and tapping it does nothing)

### Related issues

- Fixes #1884 

### Backwards compatibility

yes
